### DISCUSSION TITLE
How to use func(*pgxmock.pgxmock) error in NewConn or NewPool functions ?

### DISCUSSION THREAD

**Discussion by nononsensecode** [pashagolub/pgxmock#106] [Q&A]
Hello,
The method signature of the function `NewPool/NewConn` includes an `options` parameter of type `func(*pgxmock.pgxmock) error`. I am a bit confused as `pgxmock.pgxmock` is not an exported type, how can we use that ?

**Comment by pashagolub** [comment]
Hey, it's been awhile since the start of this discussion. I hope everything is fixed and is working. Please, reopen it if you think we may somehow improve the source code.
